proposals/infra/INF-0002-root-signature-in-clang.md

```diff
@@ -304,15 +440,92 @@ elimation has completed.
 
 ### Validations in Sema
 
-TODO
+#### All the values should be legal.
+
+Most values like ShaderVisibility/ParameterType are covered by syntactical checks in Sema.
+Only list special rule here.
+
+##### DescriptorRangeFlags
+
+###### Sampler
+- None
+- DescriptorsVolatile
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS
+
+###### Resource
+- None
+- DESCRIPTORS_VOLATILE,
+- DATA_VOLATILE,
+- DATA_STATIC,
+- DATA_STATIC_WHILE_SET_AT_EXECUTE,
+- DESCRIPTORS_VOLATILE | DATA_VOLATILE,
+- DESCRIPTORS_VOLATILE | DATA_STATIC_WHILE_SET_AT_EXECUTE,
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS,
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_VOLATILE,
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_STATIC,
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_STATIC_WHILE_SET_AT_EXECUTE,
+
+##### StaticSampler
+- Max/MinLOD cannot be NaN.
+- MaxAnisotropy cannot exceed 16.
+- MipLODBias must be within range of [-16, 15.99].
+
+##### Register Space
+
+  The range 0xFFFFFFF0 to 0xFFFFFFFF is reserved.
+```
+  "CBV(b0, space=4294967295)" is invalid due to the use of reserved space 0xFFFFFFFF.
+```
+
+#### Resource ranges must not overlap.
+```
+  "CBV(b2), DescriptorTable(CBV(b0, numDescriptors=5))" will result in an error
+  due to overlapping at b2.
+```
 
 ### Metadata Schema
 
 TODO
 
-### Validations during Codegen
+### Validations during DXIL generation
+  [Validations In Sema](#validations-in-sema)
+#### All the things validated in Sema.
+  In DXIL generation, values like ShaderVisibility/ParameterType need to be
```

Comment:
Let's just list all of the things "like ShaderVisibility/ParameterType" here.

Reply:
Updated.